packages/services/f-braze-adapter/src/services/BrazeConsumer.js

```diff
@@ -0,0 +1,114 @@
+            'Home_Promotion_Card_2',
+            'Post_Order_Card_1',
+            'Promotion_Card_1',
+            'Promotion_Card_2'
+        ];
+
+        this.enabledCardTypes = enabledCardTypes || this.defaultEnabledCardTypes;
+
+        this.customFilters = customFilters;
+
+        this.inAppMessagesCallbacks = interceptInAppMessages;
+
+        this.inAppMessageClickEventsCallbacks = interceptInAppMessageClickEvents;
+
+        this.callbacks = callbacks;
+
+        this.defaultFiltersInOrder = [
+            sortByCardOrder,
+            cards => filterByEnabledCardTypes(cards, this.enabledCardTypes),
+            cards => filterByBrands(cards, this.brands),
+            filterByCurrentlyActive
+        ];
+
+        this._cards = [];
+
+        this.brands = [];
+
+        // Here we just resolve a promise on brand to allow passing in a promise that resolves to a array of strings
+        // and also accepts a array strings
+        Promise.resolve(brands).then(brandsList => {
+            this.brands = uniq([...this.brands, ...brandsList]);
+            // check to see if cards has already been set if so re call all the content card callbacks
+            if (this._cards.length > 0) {
+                this.getContentCardCallbacks().forEach(callback => {
+                    callback(this._cards);
+                });
+            }
+        });
+
+        /**
+         * Creates a function that takes cards as an argument and curries over the filters in both default
+         * filters and custom filters
+         * @type {(function(*=): *)|*}
+         */
+        this.applyFilterMethods = pipe(
+            ...this.defaultFiltersInOrder,
+            ...this.customFilters
+        );
+    }
+
+    /**
+     * Function that loops over all inAppMessageClickEventsCallbacks and calls them
+     * @returns {*[]}
+     */
+    getInAppMessageClickEventCallbacks () {
+        return Object.keys(this.inAppMessageClickEventsCallbacks)
+            .map(key => this.inAppMessageClickEventsCallbacks[key]);
+    }
+
+    /**
+     * Function that loops over all inAppMessagesCallbacks and calls them
+     * @returns {*[]}
+     */
+    getInAppMessagesCallbacks () {
+        return Object.keys(this.inAppMessagesCallbacks)
+            .map(key => this.inAppMessagesCallbacks[key]);
+    }
+
+    /**
+     * Method takes all the consumer callbacks for content cards and wraps each of them in an anonymous function.
+     * This enables the ability to set the cards on the consumer class, which in turn applies any filter functions to
+     * the returned cards. Once the filters have been applied then it calls the callbacks.
+     * @returns {(function(*): void)[]}
+     */
+    getContentCardCallbacks () {
+        const _this = this;
+        // eslint-disable-next-line func-names
+        return Object.keys(this.callbacks).map(key => function (cards) {
+            _this.cards = cards;
+            _this.callbacks[key](_this._cards);
+        });
```

Comment:
You can switch this around with the arrow function inside the keyword function, you can use the [`thisArg` map parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#parameters) to avoid having to reassign `this` to `_this`. I'd also say that naming functions would be better for traversing stack traces for debugging:
```suggestion
        return Object.keys(this.callbacks).map(function mapContentCardCallbacks(key) {
            return (cards) => {
                this.cards = cards;
                this.callbacks[key](this._cards);
            }
        }, this);
```

Reply:
Thanks good suggestion